sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -1005,4 +1047,529 @@ export class DataLakeFileClient extends DataLakePathClient {
+        throw new RangeError(
+          `initialTransferSize option must be >= 1 and <= ${FILE_MAX_INITIAL_TRANSFER_SIZE}`
+        );
+      }
+
+      // When buffer length <= initialTransferSize, this method will use one append/flush call to finish the upload.
+      if (size <= options.initialTransferSize) {
+        await this.append(() => fs.createReadStream(filePath), 0, size, {
+          abortSignal: options.abortSignal,
+          conditions: options.conditions,
+          tracingOptions: { ...options!.tracingOptions, spanOptions }
+        });
+
+        return await this.flush(size, {
+          abortSignal: options.abortSignal,
+          conditions: options.conditions,
+          close: options.close,
+          pathHttpHeaders: options.pathHttpHeaders,
+          tracingOptions: { ...options!.tracingOptions, spanOptions }
+        })
+      }
+
+      const numBlocks: number = Math.floor((size - 1) / options.chunkSize) + 1;
+      let transferProgress: number = 0;
+      const batch = new Batch(options.maxConcurrency);
+
+      for (let i = 0; i < numBlocks; i++) {
+        batch.addOperation(
+          async (): Promise<any> => {
+            const start = options.chunkSize! * i;
+            const end = i === numBlocks - 1 ? size : start + options.chunkSize!;
+            const contentLength = end - start;
+            await this.append(
+              () => fs.createReadStream(filePath, {
+                autoClose: true,
+                end: end - 1,
+                start
+              }),
+              start,
+              contentLength,
+              {
+                abortSignal: options.abortSignal,
+                conditions: options.conditions,
+                tracingOptions: { ...options!.tracingOptions, spanOptions }
+              }
+            );
+
+            transferProgress += contentLength;
+            if (options.onProgress) {
+              options.onProgress({ loadedBytes: transferProgress });
+            }
+          }
+        );
+      }
+      await batch.do();
+
+      return await this.flush(size, {
+        abortSignal: options.abortSignal,
+        conditions: options.conditions,
+        close: options.close,
+        pathHttpHeaders: options.pathHttpHeaders,
+        tracingOptions: { ...options!.tracingOptions, spanOptions }
+      });
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Uploads a Buffer(Node.js)/Blob/ArrayBuffer/ArrayBufferView to a File.
+   *
+   * @param {Buffer | Blob | ArrayBuffer | ArrayBufferView} data Buffer(Node), Blob, ArrayBuffer or ArrayBufferView
+   * @param {FileParallelUploadOptions} [options]
+   * @returns {Promise<FileUploadResponse>}
+   * @memberof DataLakeFileClient
+   */
+  public async upload(
+    data: Buffer | Blob | ArrayBuffer | ArrayBufferView,
```

Comment:
What should we do when an empty data is provided?
Calling `append` with data length equals 0 would return an invalid header error. I think we have three options:
1. create the file but do not append or flush
   a. throw error
   b. let it pass
2. do not create the file, just throw an error

I prefer the 1.b. What do you think? @XiaoningLiu @jiacfan

Reply:
.NET is not any special handling, effectively 1.a.